internal/app/handler/post_short_link_handler.go

```diff
@@ -0,0 +1,50 @@
+package handler
+
+import (
+	"github.com/gin-gonic/gin"
+	"io"
+	"net/http"
+	"sanbright/go_shortener/internal/app/service"
+	"strings"
+)
+
+type PostShortLinkHandler struct {
+	service *service.ShortLinkService
+	baseURL string
+}
+
+func NewPostShortLinkHandler(service *service.ShortLinkService, baseURL string) *PostShortLinkHandler {
+	return &PostShortLinkHandler{service: service, baseURL: baseURL}
+}
+
+func (handler *PostShortLinkHandler) Handle(ctx *gin.Context) {
+	if ctx.Request.Method != http.MethodPost {
+		ctx.String(http.StatusMethodNotAllowed, "Method not allowed!")
+		ctx.Abort()
+		return
+	}
+
+	uri := strings.TrimLeft(ctx.Request.RequestURI, "/")
+	if len(uri) > 0 {
+		ctx.String(http.StatusBadRequest, "Not found url")
+		return
+	}
+
+	url, err := io.ReadAll(ctx.Request.Body)
+	if err != nil {
+		ctx.String(http.StatusBadRequest, "%s", err.Error())
+		ctx.Abort()
+		return
+	}
```

Comment:
Стоит закрывать `ctx.Request.Body` после чтения.
```suggestion
	url, err := io.ReadAll(ctx.Request.Body)
	if err != nil {
		ctx.String(http.StatusBadRequest, "%s", err.Error())
		ctx.Abort()
		return
	}
	defer ctx.Request.Body.Close()
```